lib/package-data.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+const request = require('request-promise-native');
+
+/**
+ * Class representing Bower package data.
+ */
+module.exports = class PackageData {
+
+	/**
+	 * Create a package data object.
+	 * @param {Object} options - The package data options.
+	 * @param {String} options.packageDataStore - The base URL of the service where initial package data will be loaded from.
+	 * @throws {TypeError} Will throw if any options are invalid.
+	 */
+	constructor(options = {}) {
+		this.options = options;
+		this.log = options.log;
+
+		if (typeof options.packageDataStore !== 'string') {
+			throw new TypeError('The packageDataStore option must be a string');
+		}
+		options.packageDataStore = options.packageDataStore.replace(/\/$/, '');
+
+		this.initialLoadComplete = false;
+		this.data = null;
+	}
+
+	/**
+	 * Load initial package data into the object, using the package data store and GitHub.
+	 */
+	loadInitialData() {
+		return Promise.all([
+			this.loadFromPackageDataStore(),
+			this.loadFromGitHub()
+		])
+		.then(() => {
+			this.initialLoadComplete = true;
+		});
+	}
+
+	/**
+	 * Load package data from the package data store and cache it in memory.
+	 * If data is already available then the result of this call will be ignored.
+	 */
+	loadFromPackageDataStore() {
+		return request({
+			uri: `${this.options.packageDataStore}/packages.json`,
```

Comment:
I was thinking it made sense for the option to be set to the base path rather than the specific URL so that if we need to add more stuff there in future we can with minimal effort. Maybe overthinking.

How about if I add this to the constructor?

```js
this.packageDataStorePath = `${this.options.packageDataStore}/packages.json`;
```